src/main/kotlin/net/maizegenetics/phgv2/brapi/service/VariantsService.kt

```diff
@@ -0,0 +1,204 @@
+                myLogger.error("VariantsService:init - please ensure CreateRefVcf has been run and the bed file is in ${tiledbURI}/reference/")
+
+            } else {
+                val bedFile = bedFileList[0]
+                val referenceRanges = createRefRangesForCache(bedFile) as ArrayList<ReferenceRange>
+                if (referenceRanges != null) {
+                    // This list of reference ranges as stored with key "haplotype"
+                    variantCache.put("all",referenceRanges)
+                    val ehCacheStat = statisticsVariantsService.getCacheStatistics("variantCache");
+                    // using println until we get myLogger working
+                    println("variant (reference Range) cache size: ${ehCacheStat.getTierStatistics().get("OnHeap")?.getMappings()}")
+                    myLogger.info("variant (reference Range) cache size: ${ehCacheStat.getTierStatistics().get("OnHeap")?.getMappings()}")
+                }
+                println("VariantsService:init -  heapSize after create variants cache:")
+                //Sizeof.printMemoryUse()
+            }
+        }
+
+        // if no reference ranges were found, return empty list
+        if (referenceRanges == null) {
+            return Pair(TokenPagination(pageSize = pageSize, nextPageToken = null, currentPageToken = currentPageToken.toString(), totalCount = 0), emptyList())
+        }
+        // This calculation rounds up
+        val totalPages = (referenceRanges.size + pageSize - 1)/pageSize
+        if (currentPageToken >= referenceRanges.size) {
+            return Pair(TokenPagination(pageSize = pageSize, nextPageToken = null, currentPageToken = currentPageToken.toString(), totalCount = totalPages), emptyList())
+        }
+
+        // Return paginated list of variants.
+        // convert the ArrayList to a list then sort it
+        // With this, we can use the index into the list as the page tokens
+        val sortedRangeList = referenceRanges.toList()
+        Collections.sort(sortedRangeList)
+
+        val startId = currentPageToken-1 // because tokens are 1 based, but list is 0-based
+        val endId = if(startId + pageSize  < referenceRanges.size) startId + pageSize else referenceRanges.size  // endId is exclusive
+
+        val totalCount = referenceRanges.size
+
+        // Pull from the sortedRangeList just those list entries that are between the startId and endId,
+        // and write them to a new list
+        println("\nLCJ VariantsService:generateVariantsListFromCache - startId: $startId, endId: $endId \n")
+        // Note: subList the endId is exclusive, accounted for when creating the endId above
+        val variants = sortedRangeList.subList(startId,endId).map { range ->
+            Variant(
+                referenceName = range.contig,
+                start = range.start,
+                end = range.end,
+                variantDbId = "${range.contig}:${range.start}:${range.end}",
```

Comment:
@lynnjo Can you do some research to make sure that ':'s are valid url characters?  It likely works, but just to verify that RESTFul doesn't assume that : means something else.